pkg/cloudproviders/aws/metadata.go

```diff
@@ -2,24 +2,62 @@ package aws
 
 import (
 	"context"
+	"encoding/base64"
+	"encoding/json"
+	"net/http"
+	"time"
 
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/ec2metadata"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/fullsailor/pkcs7"
+	"github.com/pkg/errors"
 	"github.com/stackrox/rox/generated/storage"
 	"github.com/stackrox/rox/pkg/errorhelpers"
+	"github.com/stackrox/rox/pkg/httputil/proxy"
+	"github.com/stackrox/rox/pkg/logging"
+)
+
+const (
+	timeout = 5 * time.Second
+)
+
+var (
+	log = logging.LoggerForModule()
+
+	httpClient = &http.Client{
+		Timeout:   timeout,
+		Transport: proxy.Without(),
+	}
 )
 
 // GetMetadata tries to obtain the AWS instance metadata.
 // If not on AWS, returns nil, nil.
 func GetMetadata(ctx context.Context) (*storage.ProviderMetadata, error) {
-	var verified bool
-
 	errs := errorhelpers.NewErrorList("retrieving AWS EC2 metadata")
```

Comment:
let's just move this down to where we start having multiple and default to errors.wrap